syft/frameworks/torch/tensors/interpreters/native.py

```diff
@@ -1049,19 +1047,23 @@ def decrypt(self, protocol="mpc", **kwargs):
                     private_key (phe.paillier.PaillierPrivateKey): Can be obtained using
                         ```public_key, private_key = sy.frameworks.torch.he.paillier.keygen()```
         Returns:
-            An decrypted version of the Tensor following the protocol specified
+            An decrypted version of the Tensor following the protocol guessed from its type
 
         Raises:
             NotImplementedError: If protocols other than the ones mentioned above are queried
 
         """
 
-        if protocol.lower() == "mpc":
+        protocol = kwargs.get("protocol", None)
+        if protocol:
+            warnings.warn("protocol should no longer be used in decrypt")
+
```

Comment:
Beware of the case where we have an AutogradTensor

Reply:
I thought that passing tests meant it's working fine, but yeah there was no tests for decrypting an AutogradTensor. Fixed that and added test for decrypting with/without grad